### PR TITLE
JS code

### PR DESCRIPTION
**Pull request by oOoOoOo0o**
JS implementation

### CODE REVIEW

File: OBlock/src/enrolment.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+
+function canStudentEnrollInModule(student, module) {
+    // Check if the student has paid fees
+    if (student.getPaidFees()) {
+        return false;
```
Comment (carthy27) [inline OBlock/src/enrolment.js:5]: Not sure what returning false does ?

File: OBlock/src/enrolment.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                found = true;
+                break;
+            }
+        }
+        if (!found) {
+            return false;
+        }
+    }
+
+    return true;
```
Comment (carthy27) [inline OBlock/src/enrolment.js:36]: If they all return false then return true so that the student can enroll ?

**Review by carthy27 — COMMENTED** [review]
few logic things i need confirmation on. Otherwise looks good.

File: OBlock/src/enrolment.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+
+function canStudentEnrollInModule(student, module) {
+    // Check if the student has paid fees
+    if (student.getPaidFees()) {
```
Comment (Bourkupine) [inline OBlock/src/enrolment.js]: should this be inverted

File: OBlock/src/enrolment.js (hunk 2 above)
Comment (Bourkupine) [reply OBlock/src/enrolment.js:36]: if none of the checks return false, then return true (as in the student meets the requirements to enrol, so return true)

File: OBlock/src/enrolment.js (hunk 1 above)
Comment (Bourkupine) [reply OBlock/src/enrolment.js:5]: in this case, it means the student hasn't paid their fees, in which case the student shouldn't be able to enrol.

In a more general sense (i.e. the other if statements), the student doesn't fit the requirements to enrol.

**Review by Bourkupine — CHANGES_REQUESTED** [review]
Left some comments, looks good bar a minor logic issue

File: OBlock/src/enrolment.js (hunk 3 above)
Comment (oOoOoOo0o) [reply OBlock/src/enrolment.js]: Yep, missed that

File: OBlock/src/module.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            this.requisiteModules.splice(index, 1);
+            return true;
+        }
+        return false;
+    }
+
+    // For when capacity is reduced
+    removeOverbookings() {
+        if (this.enrolledStudents.length > this.maxCapacity) {
+            this.enrolledStudents.splice(this.maxCapacity);
```
Comment (Bourkupine) [inline OBlock/src/module.js:83]: will this remove ALL students after the max capacity i.e. students 9 and 10 in the case of 10 students and max capacity of 8?
or just the element at ``maxCapacity``
(not too familiar with JS syntax)

**Review by Bourkupine — COMMENTED** [review]
just one comment, rest looks good.

File: OBlock/src/module.js (hunk 4 above)
Comment (oOoOoOo0o) [reply OBlock/src/module.js:83]: In this case it removes all elements at and after position maxCapacity

**Review by Bourkupine — APPROVED** [review]
Looks good, thanks for clearing up my comment

File: OBlock/src/enrolment.js (hunk 2 above)
Comment (carthy27) [reply OBlock/src/enrolment.js:36]: Thank u

**Review by carthy27 — APPROVED** [review]
Lgtm. Thanks for the logic breakdown @Bourkupine

**Pull request merged**